sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -41,29 +45,68 @@ class V2ExpressionBuilder(e: Expression) {
 
   private def generateExpression(expr: Expression): Option[V2Expression] = expr match {
     case Literal(value, dataType) => Some(LiteralValue(value, dataType))
```

Comment:
This should return AlwaysTrue or AlwaysFalse if the type is boolean

Reply:
OK